packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -134,6 +148,18 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
         return [];
     }
 
+    /**
+     * Check to see if one of the uris is in the workspace. If not then defer to language server activation
+     * to wait for open text document
+     */
+    protected async waitForItemInWorkspace(): Promise<boolean | TextDocument> {
+        const doesContain = await this.workspaceService.contains(this.workspaceContains);
+        if (this.documentSelector && !doesContain) {
+            return this.waitForOpenTextDocument(this.documentSelector);
```

Comment:
we could use `Promise.race` in `waitForActivation` instead

Reply:
If paths don't exist on the startup, should not we activate if they gets created later? Do you think it would be always covered by opening an editor? If it is good for your case i am fine to stick to checking only on the startup. It could be improve later, just asking.